### PR TITLE
Added Bootstrap command

### PR DESCRIPTION
**Pull request by ShaunakJoshi1407**
- This command bootstraps a single node Kubernetes cluster with the current node as the master node.
- Added Prerequisite checks for Bootstrap command to check whether the host is already attached to the PF9 control plane and is also attached to a cluster.
- Deletes the cluster created in case bootstrap command fails.
- Use Cases 
   - Bootstrap Info 
Bootstrap_Info

   - Local Bootstrap(Node and Cluster both are not present)
 
Bootstrap_No_Node_Local

   - Local Bootstrap(Node and Cluster both are not present) (with --no-prompt flag)
Bootstrap_No_Node_no-prompt_Local

  - Remote Bootstrap(Node and Cluster both are not present)
 
Bootstrap_No_Node_Remote

  - Remote Bootstrap(Node and Cluster both are not present) (with --no-prompt flag)
   Bootstrap_No_Node_no-prompt_Remote

  - Local(Node is Onboarded but not attached to a cluster)
  
Bootstrap_Node_Local

  - Remote(Node is Onboarded but not attached to a cluster)
  
Bootstrap_Node_Remote

  - Node is Onboarded and attached to a Cluster
Both_Cluster_Node_Present

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -546,3 +546,69 @@ Node decommissioned successfully
...
+
+Optional pre-requisite check(s) failed. See /home/ubuntu/pf9/log/pf9ctl-20211119.log or use --verbose for logs 
+Prep local node as master node for kubernetes cluster (y/n): y
+✓ Platform9 packages installed successfully
+✓ Initialised host successfully
+✓ Host successfully attached to the Platform9 control-plane
+✓ Cluster created successfully
+✓ Host is connected
+✓ Attached node to the cluster
+2021-11-19T07:06:58.8159Z	INFO	=======Bootstrap successfully finished========
```
Comment (anupbarve) [inline README.md:612]: This shouldn't be a log message.

File: cmd/bootstrap.go [hunk 2]
```diff
@@ -22,9 +26,7 @@ import (
 var bootstrapCmd = &cobra.Command{
 	Use:   "bootstrap",
 	Short: "Create a single node k8s cluster with your current node",
-	Long: `Bootstrap a single node Kubernetes cluster with your current
-	host as the Kubernetes node. Read more at
-	http://pf9.io/cli_clbootstrap.`,
+	Long:  `Bootstrap a single node Kubernetes cluster with your current node as the master node.`,
```
Comment (anupbarve) [inline cmd/bootstrap.go:29]: %s/your node/node

File: README.md [hunk 3]
```diff
@@ -546,3 +546,69 @@ Node decommissioned successfully
 
 ✓ Has access
 ```
+
+ **bootstrap**
+
+ ```sh
+./pf9ctl bootstrap --help
+Bootstrap a single node Kubernetes cluster with your current node as the master node.
```
Comment (anupbarve) [inline README.md:554]: %s/your current/current

File: pkg/pmk/cluster.go [hunk 4]
```diff
@@ -1,64 +1,228 @@
...
+				zap.S().Errorf("Unable to send Segment event for bootstrap node. Error: %s", err.Error())
+			}
+			zap.S().Debugf("Unable to delete the cluster")
+		} else {
+			if err = c.Segment.SendEvent("Delete Cluster(Bootstrap)", keystoneAuth, checkPass, ""); err != nil {
+				zap.S().Errorf("Unable to send Segment event for bootstrap node. Error: %s", err.Error())
+			}
+			zap.S().Debugf("Deleted the cluster successfully")
+		}
+		zap.S().Fatalf("Host is disconnected....Unable to attach this node to the cluster " + req.Name)
```
Comment (anupbarve) [inline pkg/pmk/cluster.go:110]: Fold this logic in a function.

File: pkg/pmk/cluster.go [hunk 5]
```diff
@@ -1,64 +1,228 @@
...
+		val, err := Instance.CheckExistingInstallation()
+		if err != nil {
+			//zap.S().Fatalf("Could not run command Installation")
+			zap.S().Fatalf("Error %s", err)
+		}
+
+		val1, err1 := Instance.CheckKubernetesCluster()
+		if err1 != nil {
+			//zap.S().Fatalf("Could not run command Cluster")
+			zap.S().Fatalf("Error %s", err1)
```
Comment (anupbarve) [inline pkg/pmk/cluster.go:207]: common out duplicate code which is repeated in lines 212-223

**Review by anupbarve — CHANGES_REQUESTED** [review]
Overall looks good.
Minor comments.

File: README.md (hunk 1 above)
Comment (ShaunakJoshi1407) [reply README.md:612]: Will change this to a normal println message

File: cmd/bootstrap.go (hunk 2 above)
Comment (ShaunakJoshi1407) [reply cmd/bootstrap.go:29]: will do

File: README.md (hunk 3 above)
Comment (ShaunakJoshi1407) [reply README.md:554]: will do

**Review by anupbarve — APPROVED** [review]

File: pkg/pmk/cluster.go (hunk 4 above)
Comment (ShaunakJoshi1407) [reply pkg/pmk/cluster.go]: Done

File: pkg/pmk/cluster.go (hunk 5 above)
Comment (ShaunakJoshi1407) [reply pkg/pmk/cluster.go]: Done

**Pull request merged**